src/backend/oneapi/kernel/gradient.hpp

```diff
@@ -0,0 +1,173 @@
+/*******************************************************
+ * Copyright (c) 2022, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <Param.hpp>
+#include <common/dispatch.hpp>
+#include <common/kernel_cache.hpp>
+#include <debug_oneapi.hpp>
+#include <kernel/default_config.hpp>
+
+#include <string>
+#include <vector>
+
+namespace oneapi {
+namespace kernel {
+
+template<typename T>
+using local_accessor = sycl::accessor<T, 1, sycl::access::mode::read_write,
+                                      sycl::access::target::local>;
+template<typename T>
+using read_accessor = sycl::accessor<T, 1, sycl::access::mode::read>;
+template<typename T>
+using write_accessor = sycl::accessor<T, 1, sycl::access::mode::write>;
+
+#define sidx(y, x) scratch_[((y + 1) * (TX_ + 2)) + (x + 1)]
+
+template<typename T>
+class gradientCreateKernel {
+   public:
+    gradientCreateKernel(write_accessor<T> d_grad0, const KParam grad0,
+                         write_accessor<T> d_grad1, const KParam grad1,
+                         read_accessor<T> d_in, const KParam in,
+                         const int blocksPerMatX, const int blocksPerMatY,
+                         const int TX, const int TY, local_accessor<T> scratch)
+        : d_grad0_(d_grad0)
+        , grad0_(grad0)
+        , d_grad1_(d_grad1)
+        , grad1_(grad1)
+        , d_in_(d_in)
+        , in_(in)
+        , blocksPerMatX_(blocksPerMatX)
+        , blocksPerMatY_(blocksPerMatY)
+        , TX_(TX)
+        , TY_(TY)
```

Comment:
Can't you get this informaiton from the get_work_group_size?

Reply:
fixed